src/protocol/mdns.rs

```diff
@@ -276,13 +276,14 @@ impl Mdns {
     }
 
     /// Event loop for [`Mdns`].
-    pub(crate) async fn start(mut self) -> crate::Result<()> {
+    pub(crate) async fn start(mut self) {
         tracing::debug!(target: LOG_TARGET, "starting mdns event loop");
 
-        // before starting the loop, make an initial query to the network
-        //
-        // bail early if the socket is not working
-        self.on_outbound_request().await?;
+        // Before starting the loop, make an initial query to the network
+        if let Err(error) = self.on_outbound_request().await {
+            tracing::error!(target: LOG_TARGET, ?error, "Failed to send initial mdns query. MDNS entering failure mode");
+            futures::future::pending::<()>().await;
```

Comment:
Basically execution would be stuck here forever, right? Is it a problem that `self.socket` will never be polled then below?

Reply:
Would [`tokio::time::Interval`](https://docs.rs/tokio/latest/tokio/time/struct.Interval.html) work here? It fires immediately the first time, so there will be no need to send the first outbound request manually.

The caveat is to use `Delay` [`MissedTickBehavior`](https://docs.rs/tokio/latest/tokio/time/enum.MissedTickBehavior.html), otherwise we might end up bursting many packets if for some reason other branches of `select!` take too long.

Reply:
Yep indeed, have ended up using tokio::time::Interval here 🙏